### DISCUSSION TITLE
Automatically pause replay of a log when a certain condition is satisfied

### DISCUSSION THREAD

**Discussion by ilmagico** [foxglove/community#885] [Q&A]
Would be nice to be able to replay a long log at fast speed, but have some kind of condition set (e.g. via user script) whereas if a given condition is met (e.g. a message with a given value or in a given range, or matching a regex, is received in a topic) the replay is automatically stopped and the value that triggered the condition highlighted somehow.

Alternatively, or in addition to this, would be nice to have a "search in log" function where foxglove can find all the instances of a given condition being satisfied in the log, and allow to skip directly to those timestamps.

I didn't find a way to do this with current user scripts, not sure if this is possible via an extension / custom panel, but it would be nice to have it in the core Foxglove Studio.

**Comment by jtbandes** [comment]
Duplicate of https://github.com/orgs/foxglove/discussions/230

**Reply by ilmagico** [reply]
Thanks, upvoted. I had searched in the "issues" but not in the discussions.
